packages/sdk/src/client/backend.ts

```diff
@@ -276,20 +276,36 @@ function processCampaign(backendCampaign: BackendCampaign): Campaign {
         });
     }
 
-    return <Campaign>{
-        ...backendCampaign,
-        chainId: toNumberOrNull(backendCampaign.chainId),
+    const { specification, ...rest } = backendCampaign;
+
+    const campaign: Campaign = {
+        ...rest,
+        chainId: Number(backendCampaign.chainId),
         status,
         pool: {
             ...backendCampaign.pool,
+            // FIXME: we should return the pool chain id from the backend
+            chainId: Number(backendCampaign.chainId),
             amm: backendCampaign.pool.amm as SupportedAmm,
             fee: toNumberOrNull(backendCampaign.pool.fee),
             tvl: toNumberOrNull(backendCampaign.pool.tvl),
             token0: processErc20Token(backendCampaign.pool.token0),
             token1: processErc20Token(backendCampaign.pool.token1),
         },
+        apr: toNumberOrNull(backendCampaign.apr),
         rewards,
+        whitelist: null,
```

Comment:
Do you think it's best to do it this way instead of having the `specification` field and define a specification type?

Reply:
Yeah, I agree, it's cleaner to just have a `specification` field in the `campaign`, addressed in 2f8d27fdcd5a228a9dd3b57f094e2adfd960ef24.